cupyx/scipy/ndimage/filters.py

```diff
@@ -148,6 +308,86 @@ def _generate_boundary_condition_ops(mode, ix, xsize):
     return ops
 
 
+def _get_nd_kernel(name, pre, found, post, mode, wshape, int_type,
+                   origins, cval, preamble='', options=(), has_weights=True):
+    ndim = len(wshape)
+    in_params = 'raw X x, raw W w'
+    out_params = 'Y y'
+
+    inds = _generate_indices_ops(
+        ndim, int_type, 'xsize_{j}',
+        [' - {}'.format(wshape[j]//2 + origins[j]) for j in range(ndim)])
+    sizes = ['{type} xsize_{j}=x.shape()[{j}], xstride_{j}=x.strides()[{j}];'.
+             format(j=j, type=int_type) for j in range(ndim)]
+    cond = ' || '.join(['(ix_{0} < 0)'.format(j) for j in range(ndim)])
+    expr = ' + '.join(['ix_{0}'.format(j) for j in range(ndim)])
+
+    if has_weights:
+        weights_init = 'const W* weights = (const W*)&w[0];\nint iw = 0;'
+        weights_check = 'W wval = weights[iw++];\nif (wval)'
+    else:
+        in_params = 'raw X x'
+        weights_init = weights_check = ''
+
+    loops = []
+    for j in range(ndim):
+        if wshape[j] == 1:
+            loops.append('{{ {type} ix_{j} = ind_{j} * xstride_{j};'.
+                         format(j=j, type=int_type))
+        else:
+            boundary = _generate_boundary_condition_ops(
+                mode, 'ix_{}'.format(j), 'xsize_{}'.format(j))
+            loops.append('''
+    for (int iw_{j} = 0; iw_{j} < {wsize}; iw_{j}++)
+    {{
+        {type} ix_{j} = ind_{j} + iw_{j};
+        {boundary}
+        ix_{j} *= xstride_{j};
+        '''.format(j=j, wsize=wshape[j], boundary=boundary, type=int_type))
+
+    value = '(*(X*)&data[{expr}])'.format(expr=expr)
+    if mode == 'constant':
+        value = '(({cond}) ? (X){cval} : {value})'.format(
+            cond=cond, cval=cval, value=value)
+    found = found.format(value=value)
+
+    operation = '''
+    {sizes}
+    {inds}
+    // don't use a CArray for indexing (faster to deal with indexing ourselves)
```

Comment:
I think this should be solved now in the master branch. it would be great if you can measure the performance difference with this and without this enabled.